src/Containers/Microsoft.NET.Build.Containers/Tasks/ComputeDotnetBaseImageAndTag.cs

```diff
@@ -93,16 +100,27 @@ public ComputeDotnetBaseImageAndTag()
         ContainerFamily = "";
         FrameworkReferences = [];
         TargetRuntimeIdentifier = "";
+        UserBaseImage = "";
     }
 
     public override bool Execute()
     {
-        var defaultRegistry = "mcr.microsoft.com";
-        if (ComputeRepositoryAndTag(out var repository, out var tag))
+        if (!RequiresInference)
         {
-            ComputedContainerBaseImage = $"{defaultRegistry}/{repository}:{tag}";
+            ComputedContainerBaseImage = UserBaseImage;
+            LogNoInferencePerformedTelemetry();
+            return true;
+        }
+        else
+        {
+            var defaultRegistry = "mcr.microsoft.com";
```

Comment:
nit: Maybe it would be better to define constant for well known registries. I saw using this URI in multiple places.